Sources/Workspace/Workspace.swift

```diff
@@ -49,44 +49,59 @@ public struct PackageFetchDetails {
 
 /// The delegate interface used by the workspace to report status information.
 public protocol WorkspaceDelegate: AnyObject {
-    /// The workspace is about to load a package manifest (which might be in the cache, or might need to be parsed). Note that this does not include speculative loading of manifests that may occr during dependency resolution; rather, it includes only the final manifest loading that happens after a particular package version has been checked out into a working directory.
+    /// The workspace is about to load a package manifest (which might be in the cache, or might need to be parsed). Note that this does not include speculative loading of manifests that may occur during
+    /// dependency resolution; rather, it includes only the final manifest loading that happens after a particular package version has been checked out into a working directory.
     func willLoadManifest(packagePath: AbsolutePath, url: String, version: Version?, packageKind: PackageReference.Kind)
     /// The workspace has loaded a package manifest, either successfully or not. The manifest is nil if an error occurs, in which case there will also be at least one error in the list of diagnostics (there may be warnings even if a manifest is loaded successfully).
     func didLoadManifest(packagePath: AbsolutePath, url: String, version: Version?, packageKind: PackageReference.Kind, manifest: Manifest?, diagnostics: [Basics.Diagnostic])
 
     /// The workspace has started fetching this repository.
     // deprecated 01/2022, remove once clients moved over
     func fetchingWillBegin(repository: String, fetchDetails: RepositoryManager.FetchDetails?)
-    func willFetchPackage(package: String, fetchDetails: PackageFetchDetails)
+    func willFetchPackage(package: PackageIdentity, packageLocation: String?, fetchDetails: PackageFetchDetails)
     /// The workspace has finished fetching this repository.
     // deprecated 01/2022, remove once clients moved over
     func fetchingDidFinish(repository: String, fetchDetails: RepositoryManager.FetchDetails?, diagnostic: Basics.Diagnostic?, duration: DispatchTimeInterval)
-    func didFetchPackage(package: String, result: Result<PackageFetchDetails, Error>, duration: DispatchTimeInterval)
+    func didFetchPackage(package: PackageIdentity, packageLocation: String?, result: Result<PackageFetchDetails, Error>, duration: DispatchTimeInterval)
     /// Called every time the progress of the git fetch operation updates.
     // deprecated 01/2022, remove once clients moved over
     func fetchingRepository(from repository: String, objectsFetched: Int, totalObjectsToFetch: Int)
-    func fetchingPackage(package: String, progress: Int64, total: Int64?)
+    func fetchingPackage(package: PackageIdentity, packageLocation: String?, progress: Int64, total: Int64?)
 
     /// The workspace has started updating this repository.
+    // deprecated 01/2022, remove once clients moved over
```

Comment:
I wonder if these deprecations are needed — when a client of libSwiftPM pulls the revision that has this change, they should be able to easily switch over.  There have been a lot more "breaky" changes than this one recently (workspace refactoring etc).